Allow user to specify their instance's URL.

agavgavi opened the pull request:
As the title says, user can specify their instance url dynamically instead of only using config.

It requires the entire URL to be entered, http, and trailing slash. URLComponents could fix this but they are such a headache it's not worth it.

tebertt reviewed (changes requested):
The URL text field should have auto correction/capitalization turned off, other than that everything looks good

agavgavi commented:
> The URL text field should have auto correction/capitalization turned off, other than that everything looks good

Fixed, also added it so the username field is autoselected when you go to that page, same with ticket title when you are creating/editing.
